source/postprocess/rotation_statistics.cc

```diff
@@ -0,0 +1,260 @@
+#include <deal.II/fe/fe_values.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace
+    {
+      void add_scientific_column(const std::string &name,
+                                 const double value,
+                                 TableHandler &statistics)
+      {
+        statistics.add_value(name,value);
+        statistics.set_precision (name, 8);
+        statistics.set_scientific (name, true);
+      }
+    }
+
+    template <int dim>
+    std::pair<std::string,std::string>
+    RotationStatistics<dim>::execute (TableHandler &statistics)
+    {
+      const QGauss<dim> quadrature_formula (this->get_fe()
+                                            .base_element(this->introspection().base_elements.velocities).degree+1);
+
+      FEValues<dim> fe_values (this->get_mapping(),
+                               this->get_fe(),
+                               quadrature_formula,
+                               update_values   |
+                               update_gradients |
+                               update_quadrature_points |
+                               update_JxW_values);
+
+      MaterialModel::MaterialModelInputs<dim> in(fe_values.n_quadrature_points, this->n_compositional_fields());
+      MaterialModel::MaterialModelOutputs<dim> out(fe_values.n_quadrature_points, this->n_compositional_fields());
+
+      Tensor<1,dim> local_angular_momentum;
+      double local_scalar_angular_momentum = 0.0;
+
+      SymmetricTensor<2,dim> local_moment_of_inertia;
+      double local_scalar_moment_of_inertia = 0.0;
+
+      typename DoFHandler<dim>::active_cell_iterator
+      cell = this->get_dof_handler().begin_active(),
+      endc = this->get_dof_handler().end();
+      for (; cell!=endc; ++cell)
+        if (cell->is_locally_owned())
+          {
+            fe_values.reinit (cell);
+
+            // Set use_strain_rates to false since we don't need viscosity
+            in.reinit(fe_values, cell, this->introspection(), this->get_solution(), false);
+            if (!use_constant_density)
+              {
+                this->get_material_model().evaluate(in, out);
+              }
+
+            for (unsigned int q = 0; q < fe_values.n_quadrature_points; ++q)
+              {
+                const double rho = (use_constant_density)
+                                   ?
+                                   1.0
+                                   :
+                                   out.densities[q];
+
+                if (dim == 2)
+                  {
+                    // Get the velocity perpendicular to the position vector
+                    const Tensor<1,dim> r_perp = cross_product_2d(in.position[q]);
+                    // calculate a signed scalar angular momentum
+                    local_scalar_angular_momentum += in.velocity[q] * r_perp * rho * fe_values.JxW(q);
+                    local_scalar_moment_of_inertia += in.position[q].norm_square() * rho * fe_values.JxW(q);
```

Comment:
Can you leave a comment here that states that you are making the assumption (see the manual) that the 2d geometry is a cross-section through an infinite domain in z direction, with a zero z-velocity. As a consequence, the angular momentum you compute here is simply the zz-component of the full 3d tensor, and is consequently a scalar. (or attach that to the declaration of the variable.)